lib/wallet/error.rs

```diff
@@ -332,9 +796,317 @@ where
+    #[error("validator is not synced")]
+    ValidatorNotSynced,
+}
+
+impl ToStatus for GenerateBlock {
+    fn builder(&self) -> StatusBuilder {
+        match self {
+            Self::GenerateCoinbaseTxouts(err) => err.builder(),
+            Self::GenerateSignetBlock(err) => err.builder(),
+            Self::Mine(err) => err.builder(),
+            Self::SelectBlockTxs(err) => err.builder(),
+            Self::TryGetMainchainTip(err) => err.builder(),
+            Self::DeleteBmmRequests(_) | Self::ValidatorNotSynced => StatusBuilder::new(self),
+        }
+    }
+}
+
+#[derive(Diagnostic, Debug, Error)]
+pub enum GetWalletBalance {
+    #[error(transparent)]
+    NotSynced(#[from] NotSynced),
+    #[error(transparent)]
+    NotUnlocked(#[from] NotUnlocked),
+}
+
+impl ToStatus for GetWalletBalance {
+    fn builder(&self) -> StatusBuilder {
+        match self {
+            Self::NotSynced(err) => err.builder(),
+            Self::NotUnlocked(err) => err.builder(),
+        }
+    }
+}
+
+#[derive(Diagnostic, Debug, Error)]
+pub enum ListWalletTransactions {
+    #[error(transparent)]
+    BitcoinCoreRPC(#[from] BitcoinCoreRPC),
+    #[error(transparent)]
+    DeserializeHex(#[from] bitcoin::consensus::encode::FromHexError),
+    #[error(transparent)]
+    NotUnlocked(#[from] NotUnlocked),
+}
+
+impl ToStatus for ListWalletTransactions {
+    fn builder(&self) -> StatusBuilder {
+        match self {
+            Self::BitcoinCoreRPC(err) => err.builder(),
+            Self::DeserializeHex(err) => StatusBuilder::new(err),
+            Self::NotUnlocked(err) => err.builder(),
+        }
+    }
+}
+
+#[derive(Diagnostic, Debug, Error)]
+pub enum CreateSendPsbt {
+    #[error(transparent)]
+    CreateTx(#[from] bdk_wallet::error::CreateTxError),
+    #[error(transparent)]
+    NotUnlocked(#[from] NotUnlocked),
+    #[error(transparent)]
+    Script(#[from] bitcoin::script::PushBytesError),
+    #[diagnostic(code(create_send_transaction_add_utxo))]
+    #[error("UTXO is not in wallet (`{}:{}`)", .0.txid, .0.vout)]
+    UnknownUTXO(bitcoin::OutPoint),
+}
+
+impl ToStatus for CreateSendPsbt {
+    fn builder(&self) -> StatusBuilder {
+        match self {
+            Self::UnknownUTXO(_) => StatusBuilder::new(self).code(tonic::Code::InvalidArgument),
+            Self::NotUnlocked(err) => err.builder(),
+            Self::CreateTx(err) => StatusBuilder::new(err),
+            Self::Script(err) => StatusBuilder::new(err),
+        }
+    }
+}
+
+#[derive(Diagnostic, Debug, Error)]
+pub enum SendWalletTransaction {
+    #[error("failed to broadcast tx")]
+    BroadcastTx(#[source] jsonrpsee::core::ClientError),
+    #[error(transparent)]
+    CreateSendPsbt(#[from] CreateSendPsbt),
+    #[error(transparent)]
+    SignTransaction(#[from] WalletSignTransaction),
+    #[error("failed to broadcast send transaction (OP_DRIVECHAIN not supported by node)")]
+    OpDrivechainNotSupported,
```

Comment:
Could consider adding an explanation for how to fix this, similar to how we do with a few other obscure errors. This relates to the node not having `acceptnonstdtxn`, right?